src/main/java/org/jenkinsci/remoting/protocol/impl/BIONetworkLayer.java

```diff
@@ -248,10 +249,23 @@ public void run() {
                         } catch (ClosedChannelException e) {
                             onRecvClosed();
                             return;
-                        } catch (EOFException e) {
+                        } catch (IOException e) {
```

Comment:
We were catching the `IOException` anyway... but I agree that we should be just looping for a `SocketTimeoutException`

Likely not an issue for most cases as we use the NIOEngine exclusively for most code paths... but yes it would be a good idea to fix anyway